### ISSUE TITLE
Deleting multiple (interrelated) organizations fails

### ISSUE THREAD

**Issue by ymc-siwi** [sulu/sulu#6600]
Deleting multiple (interrelated) organizations fails

| Q | A
| --- | ---
| Sulu Version | 2.4.2
| PHP Version | 8.1.3
| DB Version | MySQL 5.7.36
| Browser Version | Chrome 100.0.4896.127

# Actual Behavior

- Deleting multiple, interrelated organizations at the same time will fail with unexpected error (tested relation: "Parent Company")
- After failure, the list-view of organizations is not reloaded causing subsequent editorial errors (successfully deleted organizations are still visible, selection  of items in list is not updated)
- Not reproducible when deleting only one organization

# Expected Behavior

- Deleting interrelated organizations does not fail
- List view and selection are updated in case of an occuring error

# Steps to Reproduce

- Log into admin using a user that has sufficient rights to manage contacts/organizations
- Open account list view (/admin/#/accounts)
- Select multiple organizations (e.g. 10-20 items); Selection should contain organizations that have Parent Company relations (idAccountsParent) to other organizations within the selection
- Delete selected organizations via option "Delete" in the menu bar
- Sulu will trigger parallel deletion requests for every selected item in the background
- Deletion will fail with error message "Error - An unexpected error occured while deleting" in case that any of the deletion requests is failing; 2 types of errors will occur for the background requests:
1.) ```Doctrine\DBAL\Exception\DeadlockException An exception occurred while executing a query: SQLSTATE[40001]: Serialization failure: 1213 Deadlock found when trying to get lock; try restarting transaction```
2.) ```Doctrine\DBAL\Exception\ForeignKeyConstraintViolationException
An exception occurred while executing a query: SQLSTATE[23000]: Integrity constraint violation: 1451 Cannot delete or update a parent row: a foreign key constraint fails (`blk_sulu`.`co_accounts`, CONSTRAINT `FK_805CD14AC9171171` FOREIGN KEY (`idAccountsParent`) REFERENCES `co_accounts` (`id`))```
- List view is not updated and will show already deleted datasets leading to consequential errors when working with this list as an editor

# Possible Solutions

- ```sulu/src/Sulu/Bundle/ContactBundle/Resources/config/doctrine/Account.orm.xml```, line 69:
```on-delete="SET NULL"``` is missing; currently defaulting to ```on-delete="RESTRICT"``` in the database causing the failing foreign key contraints (successfully deleted datasets will leave a restricted reference)
- Scheduling or serial processing of deletion requests; Parallel deletion of related organizations will cause the observed deadlocks between datasets
- Reload list view of organizations in case that unexpected errors occur
- Possible quickfix for the time being: Disallow the deletion of multiple organizations at the same time

**Comment by alexander-schranz** [comment]
@ymc-siwi Thank you for reporting. The `on-delete="SET NULL"` totally make sense, is that enough to fix the issue? Do you want to create a Pull Request for it? (`2.3.` Branch is currently still supported for bugfixes.)

**Comment by ymc-siwi** [comment]
@alexander-schranz Sure, will prepare a pull request for the on-delete-case. Unfortunately, this will only solve the first issue (failing foreign key contraints). Deadlocks and faulty lists after failing deletion will remain an issue.

**Comment by ymc-siwi** [comment]
@alexander-schranz -> https://github.com/sulu/sulu/pull/6605

**Comment by alexander-schranz** [comment]
closed via #6605

**Comment by rogamoore** [comment]
Sorry for posting on an old issue, but I think part of the issue is still valid.
Basically if you select around 10 items (one page) or more for deletion you will always run into an error, e.g.
```
Doctrine\DBAL\Exception\DeadlockException An exception occurred while executing a query: SQLSTATE[40001]: Serialization failure: 1213 Deadlock found when trying to get lock; try restarting transaction
```
The UI will show an error message, but not update the list.
